application/clicommands/CheckCommand.php

```diff
@@ -66,42 +67,57 @@ public function hostAction()
             exit(3);
         }
 
-        $dbTool = new DbTool($this->getDb());
-        $targets = (new Select())
-            ->from('x509_target t')
-            ->columns([
-                't.port',
-                'cc.valid',
-                'cc.invalid_reason',
-                'c.subject',
-                'self_signed'   => 'COALESCE(ci.self_signed, c.self_signed)',
-                'valid_from'    => (new Select())
-                    ->from('x509_certificate_chain_link xccl')
-                    ->columns('MAX(GREATEST(xc.valid_from, xci.valid_from))')
-                    ->join('x509_certificate xc', 'xc.id = xccl.certificate_id')
-                    ->join('x509_certificate xci', 'xci.subject_hash = xc.issuer_hash')
-                    ->where('xccl.certificate_chain_id = cc.id'),
-                'valid_to'      => (new Select())
-                    ->from('x509_certificate_chain_link xccl')
-                    ->columns('MIN(LEAST(xc.valid_to, xci.valid_to))')
-                    ->join('x509_certificate xc', 'xc.id = xccl.certificate_id')
-                    ->join('x509_certificate xci', 'xci.subject_hash = xc.issuer_hash')
-                    ->where('xccl.certificate_chain_id = cc.id')
+        $conn = $this->getDb();
+        $targets = X509Target::on($conn)->with([
+            'chain',
+            'chain.certificate',
+            'chain.certificate.issuer_certificate'
+        ]);
+
+        $targets->getWith()['target.chain.certificate.issuer_certificate']->setJoinType('LEFT');
+
+        $targets->columns([
+            'port',
+            'chain.valid',
+            'chain.invalid_reason',
+            'subject'     => 'chain.certificate.subject',
+            'self_signed' => new Expression('COALESCE(%s, %s)', [
+                'chain.certificate.issuer_certificate.self_signed',
+                'chain.certificate.self_signed'
+            ])
+        ]);
+
+        // Sub queries for (valid_from, valid_to) columns
```

Comment:
I'm not sure what exactly you want different, but how do you want the ORM to natively build this for me? The only thing I could do is instead of manually creating the subqueries, replace them by `Query::createSubQuery()`. I would still have to manually assemble the subqueries before I can use them for the columns.

https://github.com/Icinga/icingadb-web/blob/master/library/Icingadb/Model/Behavior/FlattenedObjectVars.php#L45-L55

Reply:
I don't have anything specific in mind, but find it wrong to have to call `dump()`.

Reply:
I could also assemble the query manually, though, so I could just copy what `dump()` does!

Reply:
Hmm, let's leave it as it is then.